tests/test_queue_model.py

```diff
@@ -476,116 +536,164 @@ def test_make_full_path(self):
+        # trigger a queue event.
         self.model.move(sal_index=i0,
                         location=Location.AFTER,
                         location_sal_index=i0)
         await self.assert_next_queue(sal_indices=[i0+2, i0+1, i0])
 
-        # move i0+1 before itself: this should be a no-op
-        # but it should still output the queue event
+        # Move i0+1 before itself. This should be a no-op, but should still
+        # trigger a queue event.
         self.model.move(sal_index=i0+1,
                         location=Location.AFTER,
                         location_sal_index=i0+1)
         await self.assert_next_queue(sal_indices=[i0+2, i0+1, i0])
 
-        # try some incorrect moves
+        # Try some incorrect moves.
         with self.assertRaises(ValueError):
             self.model.move(sal_index=1234,  # no such script
                             location=Location.LAST,
                             location_sal_index=0)
-        await self.assert_next_queue(sal_indices=[i0+2, i0+1, i0])
+        await self.assert_next_queue(sal_indices=[i0+2, i0+1, i0], wait=False)
 
         with self.assertRaises(ValueError):
             self.model.move(sal_index=i0+1,
                             location=21,  # no such location
                             location_sal_index=0)
-        await self.assert_next_queue(sal_indices=[i0+2, i0+1, i0])
+        await self.assert_next_queue(sal_indices=[i0+2, i0+1, i0], wait=False)
 
         with self.assertRaises(ValueError):
             self.model.move(sal_index=i0+1,
                             location=Location.BEFORE,
                             location_sal_index=1234)  # no such script)
-        await self.assert_next_queue(sal_indices=[i0+2, i0+1, i0])
+        await self.assert_next_queue(sal_indices=[i0+2, i0+1, i0], wait=False)
 
-        # try incorrect index and the same "before" locationSalIndex
+        # Incorrect index and the same "before" locationSalIndex.
         with self.assertRaises(ValueError):
             self.model.move(sal_index=1234,
                             location=Location.BEFORE,
                             location_sal_index=1234)
-        await self.assert_next_queue(sal_indices=[i0+2, i0+1, i0])
+        await self.assert_next_queue(sal_indices=[i0+2, i0+1, i0], wait=False)
 
-        # try incorrect index and the same "after" locationSalIndex
+        # Incorrect index and the same "after" locationSalIndex.
         with self.assertRaises(ValueError):
             self.model.move(sal_index=1234,
                             location=Location.AFTER,
                             location_sal_index=1234)
-        await self.assert_next_queue(sal_indices=[i0+2, i0+1, i0])
+        await self.assert_next_queue(sal_indices=[i0+2, i0+1, i0], wait=False)
 
         await self.model.wait_terminate_all(timeout=STD_TIMEOUT)
 
+    async def test_clear_group_id(self):
+        """Test that a script at the top of the queue has its group ID cleared
```

Comment:
Silly question but is there a test to make sure that when the script is sent for execution (and so leaves the queue) that it's group is not cleared?

Reply:
Good suggestion. I will add that.